packages/block-library/src/post-date/edit.js

```diff
@@ -65,6 +65,11 @@ export default function PostDateEdit( {
 		} ),
 	} );
 
+	// ensure that the post date is not a link when the post template marks the entire post a link
+	useEffect( () => {
+		if ( isPostOneLink ) setAttributes( { isLink: false } );
```

Comment:
All these similar changes in blocks, create an `undo` history step. Besides that, if someone would toggle again the `isPostOneLink` shouldn't the attributes of other blocks become as before? There is no way to know this right now.